doc/kubernetes/container-image-management.md

```diff
@@ -0,0 +1,40 @@
+    - After we upgrade one container to a higher version via k8s, in this time, k8s is down or something wrong, container need to go back to local mode, local mode container should be the higher version other than the original version.
+    - Upgrade will bring a new version container image to sonic, after many times upgrade, the disk will be full, need to avoid this.
+- So we need to manage the container image to fix these two problems
+
+
+# How to handle the two problems
+- Tag latest
+    - Each time we do upgrade, after we think the new version is good, we tag the image version to latest. When need to go back to local, local mode will use the latest version image
+    - How to "think the new version is good", after the new version container is running, we will check the container is still running or not in ten minutes, still running mean the new version is good so that we can do  tag latest. Not running means fallback happens or something wrong, we can't tag the new version image to latest
+    - After we tag latest, we need to remove the previous stopped local container, otherwise when go back to local, systemd service will not create new local container with new version image and will start the stopped container which running on a previous version image.
+- Image Clean-Up
+    - Tag latest will trigger Clean-Up, once the tag latest successfully finished, we will do Clean-Up
+    - Clean-Up will clean up all old version images beside the last latest version image(for fallback usage)
+    - How to handle one special case
+        - After we upgrade one container from local mode(v1) to kube mode(v2) and we tag the v2 version image to latest, before we do Clean-Up we will find there is two cases for container image versions
+            - Case 1:
+                ```
+                docker-sonic-telemetry:latest(v2)
+                docker-sonic-telemetry:v1
+                sonick8scue.azurecr.io/docker-sonic-telemetry:v1
+                sonick8scue.azurecr.io/docker-sonic-telemetry:v2
+                ```
+            - Case 2:
+                ```
+                docker-sonic-telemetry:latest(v2)
+                docker-sonic-telemetry:v1
+                sonick8scue.azurecr.io/docker-sonic-telemetry:v2
+                ```
+        - For case-1, we remove sonick8scue.azurecr.io/docker-sonic-telemetry:v1 and tag docker-sonic-telemetry:v1 to sonick8scue.azurecr.io/docker-sonic-telemetry:v1 and remove docker-sonic-telemetry:v1.
+        - For case-2, we remove docker-sonic-telemetry:v1 directly.
+        - Why are there these two cases, find answer in next section.
+
+# Kube container will not replace the local container when their image version are the same.(Suppose the versions are both v1)
+- Why: when they are in the same version, local container should be more stable and no need to restart the container so no down time; when we rollout, we will enable k8s on many devices, we don't want the containers restart at the same time.
```

Comment:
this may be confusing, let's focus on one device.  Replacing the mode during join may cause service interruption.